client/src/glossary/GlossarySidebarContent.tsx

```diff
@@ -0,0 +1,213 @@
+import _ from "lodash";
+import React from "react";
+
+import "./GlossaryMenu.scss";
+import { glossaryEntryStore } from "src/models/glossary";
+
+import { lang } from "src/core/injected_build_constants";
+
+import { IconArrow } from "src/icons/icons";
+import { SearchHighlighter } from "src/search/search_utils";
+
+interface SidebarContentProps {
+  title: string | null;
+  def: string | null;
+  results: ResultProps[];
+  closeItem: () => void;
+  openItem: (key: string) => void;
+  showList: boolean;
+  query: string;
+}
+
+interface SidebarContentState {
+  title: string | null;
+  def: string | "";
+  scrollEl: string;
+}
+
+export interface ResultProps {
+  id: string;
+  title: string;
+  translation: string;
+  raw_definition: string;
+}
+
+export class SidebarContent extends React.Component<
+  SidebarContentProps,
+  SidebarContentState
+> {
+  listRef = React.createRef<HTMLDivElement>();
+  constructor(props: SidebarContentProps) {
+    super(props);
+
+    this.state = {
+      title: null,
+      def: "",
+      scrollEl: "",
+    };
+  }
+
+  static getDerivedStateFromProps(
+    nextProps: SidebarContentProps,
+    prevState: SidebarContentState
+  ) {
+    const { title: next_title, def: next_def } = nextProps;
+    const { title: prev_title } = prevState;
+
+    if (next_title !== prev_title) {
+      return {
+        title: next_title,
+        def: next_def,
+      };
+    } else {
+      return null;
+    }
+  }
+
+  closeDefinition() {
+    this.props.closeItem();
+    if (this.state.title) {
+      this.setState({
+        scrollEl: this.state.title.replace(/\s+/g, ""),
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.scrollEl) {
+      const el = document.getElementById(this.state.scrollEl);
+      const scrollDiv = document.getElementById("gloss-sidebar");
+
+      if (el && scrollDiv) {
+        scrollDiv.scrollTop = el.offsetTop;
+        el.focus();
+      }
+    }
+  }
+
+  openDefinition(item: ResultProps) {
+    this.props.openItem(item.id);
+    this.setState({
+      scrollEl: item.title.replace(/\s+/g, ""),
+    });
+  }
+
+  //place holder arguments while I get passing functions to work...
+  handleKeyPress(
+    e: React.KeyboardEvent<HTMLSpanElement>,
+    action: string,
+    item: ResultProps | null
+  ) {
+    if (e.key === "Enter") {
+      switch (action) {
+        case "close":
+          this.closeDefinition();
+          break;
+
+        case "open":
+          if (item) {
+            this.openDefinition(item);
+          }
+          break;
+      }
+    }
+  }
+
+  get_glossary_items_by_letter() {
```

Comment:
This code looks like it's copied from `glossary.js`, right? Extract it to `src/glossary/glossary_utils.ts` (along with any other common code)